src/snowflake/snowpark/_internal/utils.py

```diff
@@ -423,14 +444,19 @@ def calculate_checksum(
                 # ignore generated python files
                 if ignore_generated_py_file and file.endswith(GENERATED_PY_FILE_EXT):
                     continue
+
                 hash_algo.update(file.encode("utf8"))
+
+                filename = os.path.join(dirname, file)
+                file_size = os.path.getsize(filename)
+
+                if whole_file_hash:
+                    _hash_file(hash_algo, filename, chunk_size, whole_file_hash)
+                    current_size += file_size
                 if current_size < chunk_size:
```

Comment:
I'm curious will this `if current_size < chunk_size:` still meets the condition after the first `if whole_file_hash` meets the condition, in which case `_hash_file` will be executed twice? or is this a valid scenario?

Reply:
That's a good catch. I believe it should be an `elif`. Hashing the file a second time should break anything, but it also is wasted computation.